File: Dataset/Leetcode/train/5/167.java
 class Solution {
    public String XXX(String s) {
        if (s == null || s.equals("")) {
            return "";
        }
        int maxLen = 0;
        int start = 0, end = 0;
        for (int i = 0; i < s.length() - 1; i++) {
            // aba 这种情况
            int len1 = findMaxLen(s, i, i);
            // aa 这种情况
            int len2 = findMaxLen(s, i, i + 1);
            // aba 这种情况, 求到的len一定是个奇数
            if (len1 > maxLen) {
                maxLen = len1;
                start = i - maxLen / 2;
                end = i + maxLen / 2;
            }
            // aa 这种情况, 求到的len一定是个偶数
            if (len2 > maxLen) {
                maxLen = len2;
                start = i - maxLen / 2 + 1;
                end = i + maxLen / 2;
            }
        }
        return s.substring(start, end + 1);
    }
    
    private int findMaxLen(String s, int start, int end) {
        // 只要相等, 就不断向两边扩展
        while (start >= 0 && end < s.length() && s.charAt(start) == s.charAt(end)) {
            start--;
            end++;
        }
        return end - start - 1;
    }
}

File: Dataset/Leetcode/train/38/167.cpp
class Solution {
public:
    string XXX(int n) {
        if(n==1)
            return "1";
        if(n==2)
            return "11";
        string s1="11",s2;
        for(int i=3;i<=n;i++)
        {
            int cnt=1;
            for(int j=1;j<s1.size();j++)
            {
                if(s1[j]==s1[j-1])
                    cnt++;
                else
                {
                    s2+=cnt+'0';
                    s2+=s1[j-1];
                    cnt=1;
                }
                if(j==s1.size()-1)
                {
                    s2+=cnt+'0';
                    s2+=s1[j];
                }
            }
            s1=s2;
            s2="";
        }
        return s1;
    }
};

